### PR TITLE
spec: wrap/override fsspec httpfs

### PR DESCRIPTION
**Pull request by pmrowla**
Fixes https://github.com/iterative/dvc-objects/issues/195

### CODE REVIEW

**Comment by pmrowla** [comment]
It would probably be better for our filesystems to just wrap fsspec directly instead of using the `fs.fs` redirection that we have right now, but this fixes the issue until we can actually get around to refactoring `dvc_objects.fs`

File: dvc_http/spec.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from fsspec.implementations.http import HTTPFileSystem
+
+
+class HTTPFileSystemWrapper(HTTPFileSystem):  # pylint: disable=abstract-method
```
Comment (skshetry) [inline dvc_http/spec.py:4]: Let's keep the same name.
```suggestion
from fsspec.implementations import http


class HTTPFileSystem(http.HTTPFileSystem):  # pylint: disable=abstract-method
```

File: dvc_http/__init__.py [hunk 2]
```diff
@@ -129,28 +128,17 @@ async def get_client(self, **kwargs):
...
     def fs(self):
-        from fsspec.implementations.http import (
-            HTTPFileSystem as _HTTPFileSystem,
-        )
+        from .spec import HTTPFileSystemWrapper
 
-        return _HTTPFileSystem(get_client=self.get_client, **self.fs_args)
+        return HTTPFileSystemWrapper(
+            get_client=self.get_client,
+            upload_method=self.upload_method,
```
Comment (skshetry) [inline dvc_http/__init__.py:135]: Let's get rid of `self.upload_method` and set these in `__init__()` itself in `self.fs_args`.

```python
self.fs_args["upload_method"] = kwargs.get("method", "POST")
```

**Comment by skshetry** [comment]
> It would probably be better for our filesystems to just wrap fsspec directly instead of using the fs.fs redirection that we have right now, but this fixes the issue until we can actually get around to refactoring dvc_objects.fs

Ideally, we should just directly use fsspec-provided filesystems. There should be no wrappers.

**Review by skshetry — APPROVED** [review]
@pmrowla, I have approved the PR, I have a few minor suggestions inlined. But I think it'd be better to upstream this in fsspec's http implementation itself.

Also, I'll encourage you to open a discussion regarding batching issue in fsspec, as this `dvc-http` is not the only place this might break.
We also have similar code in `dvc-azure`, but I am not sure if that breaks anything or not.

https://github.com/iterative/dvc-azure/blob/78e3ff570d176bec954915847c403122a2b8cf95/dvc_azure/__init__.py#L178-L180

**Pull request merged**